modules/analysis/src/service/load.rs

```diff
@@ -237,9 +248,59 @@ impl AnalysisService {
                 .distinct()
                 .into_query(),
             GraphQuery::Query(query) => sbom_node::Entity::find()
-                .filtering(query.clone())?
+                .join(JoinType::Join, sbom_node::Relation::Package.def())
+                .join(JoinType::LeftJoin, sbom_package::Relation::Purl.def())
+                .join(JoinType::LeftJoin, sbom_package::Relation::Cpe.def())
+                .join(
+                    JoinType::LeftJoin,
+                    sbom_package_cpe_ref::Relation::Cpe.def(),
+                )
                 .select_only()
                 .column(sbom_node::Column::SbomId)
+                .filtering_with(
+                    query.clone(),
+                    sbom_node::Entity
+                        .columns()
+                        .add_columns(cpe::Entity.columns())
+                        .translator(|f, op, v| {
+                            match (f, op, OwnedUri::from_str(v)) {
```

Comment:
I cannot tell what the resultant sql code looks like from this code block ... it might be we need to do a pg func.

Reply:
I should also point out that the triggering of a full text search by an invalid CPE only affects what's loaded into the graph. But the querying of the loaded graph will still be accurate. The user will never see whatever false positives were initially loaded by the full-text search.

This is an extreme edge case, obviously. :wink: